File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    slist = list(s)*2
    length = len(s)
    answer = 0 
    for i in range(length) :

        comparelist = slist[i:len(s)+i]
        
        #올바른지만 보면 됨
        
        q =[]
        for j in range(len(comparelist)) :
            if len(q)>0:
                if comparelist[j] == "}" and q[-1] == "{":
                    q.pop()
                elif comparelist[j] == ")" and q[-1] == "(":
                    q.pop()
                elif comparelist[j] == "]" and q[-1] == "[":
                    q.pop()
                else :
                    q.append(comparelist[j])
            
            else : 
                q.append(comparelist[j])
        
        if len(q) == 0:
            answer +=1 
        
    
    return answer

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        StringBuilder sb = new StringBuilder(s);
        
        //sb를 돌면서, 한개씩 회전
        int answer = 0 ;
        int slen = s.length();
        for (int roll = 0 ; roll< slen; roll++)
        {
            char now = sb.charAt(0);
            // 돈 값에 대해서 검사 .
            Stack<Character> st = new Stack<Character>();
            st.push(now);
            for (int i = 1 ; i<sb.length(); i++)
            {
                Character nowchar = sb.charAt(i);
                if (nowchar == ']')
                {
                    if (!st.isEmpty() && st.peek() == '[')
                    {
                        st.pop();
                    }else
                    {
                        st.push(nowchar);
                    }
                } else if (nowchar == '}')
                {
                    if (!st.isEmpty() && st.peek() == '{')
                    {
                        st.pop();
                    }else
                    {
                        st.push(nowchar);
                    }
                } else if (nowchar == ')')
                {
                    if (!st.isEmpty() && st.peek() == '(')
                    {
                        st.pop();
                    }else
                    {
                        st.push(nowchar);
                    }
                } else
                {
                    st.push(nowchar);
                }
            }

            if(st.isEmpty())
            {
                answer +=1;
            }
            
            sb.deleteCharAt(0);
            sb.append(now);
            
        }
        
    
        return answer;
    }
}